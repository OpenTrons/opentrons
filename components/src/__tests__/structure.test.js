// structure components tests
import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import Renderer from 'react-test-renderer'

import { PageTabs, TitleBar, Card, LabeledValue, Splash, Pill } from '..'

describe('TitleBar', () => {
  it('adds an h1 with the title', () => {
    const heading = Renderer.create(<TitleBar title="hello" />).root.findByType(
      'h1'
    )

    expect(heading).toBeDefined()
    expect(heading.children).toEqual(['hello'])
  })

  it('adds an optional h2 with the subtitle', () => {
    const heading = Renderer.create(
      <TitleBar title="hello" subtitle="world" />
    ).root.findByType('h2')

    expect(heading).toBeDefined()
    expect(heading.children).toEqual(['world'])
  })

  it('add optional back button', () => {
    const onBackClick = jest.fn()
    const button = Renderer.create(
      <TitleBar title="hello" onBackClick={onBackClick} />
    ).root.findByType('button')

    button.props.onClick()
    expect(onBackClick).toHaveBeenCalled()
  })

  it('renders TitleBar without subtitle correctly', () => {
    const tree = Renderer.create(<TitleBar title="foo" />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders TitleBar with subtitle correctly', () => {
    const tree = Renderer.create(
      <TitleBar title="foo" subtitle="bar" />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders TitleBar with back button correctly', () => {
    const tree = Renderer.create(
      <TitleBar title="foo" subtitle="bar" onBackClick={() => {}} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('PageTabs', () => {
  it('renders h3 links for each page', () => {
    const pages = [
      { title: 'Page1', href: '/page1', isActive: false, isDisabled: false },
      { title: 'Page2', href: '/page2', isActive: false, isDisabled: false },
    ]

    const root = Renderer.create(
      <MemoryRouter>
        <PageTabs pages={pages} />
      </MemoryRouter>
    ).root

    const links = root.findAllByType('a')
    expect(links).toHaveLength(2)

    links.forEach((link, index) => {
      const { title, href } = pages[index]
      expect(link.props.href).toBe(href)
      expect(link.findByType('h3').children).toEqual([title])
    })
  })

  it('does not create a link if disabled', () => {
    const pages = [
      { title: 'Page1', href: '/page1', isActive: false, isDisabled: true },
    ]

    const notLink = Renderer.create(
      <MemoryRouter>
        <PageTabs pages={pages} />
      </MemoryRouter>
    ).root.findByType('span')

    expect(notLink.findByType('h3').children).toEqual([pages[0].title])
  })

  it('adds active class if active', () => {
    const pages = [
      { title: 'Page1', href: '/page1', isActive: true, isDisabled: false },
    ]

    const link = Renderer.create(
      <MemoryRouter>
        <PageTabs pages={pages} />
      </MemoryRouter>
    ).root.findByType('a')

    expect(link.props.className).toMatch(/active/)
  })

  it('renders PageTabs correctly', () => {
    const pages = [
      { title: 'Page1', href: '/page1', isActive: true, isDisabled: false },
      { title: 'Page2', href: '/page2', isActive: false, isDisabled: true },
    ]

    const tree = Renderer.create(
      <MemoryRouter>
        <PageTabs pages={pages} />
      </MemoryRouter>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('Card', () => {
  it('renders Card correctly', () => {
    const tree = Renderer.create(
      <Card title={'title'}>children children children</Card>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('LabeledValue', () => {
  it('renders LabeledValue correctly', () => {
    const tree = Renderer.create(
      <LabeledValue label={'Label'} value={'Value'} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('Splash', () => {
  it('renders correctly with no props', () => {
    const tree = Renderer.create(<Splash />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with custom props', () => {
    const tree = Renderer.create(
      <Splash iconName="flask-outline" className="swag" />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})

describe('Pill', () => {
  it('renders Pill correctly', () => {
    const tree = Renderer.create(
      <Pill color="blue" className="foo">
        Blue
      </Pill>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders Pill correctly with inverted text', () => {
    const tree = Renderer.create(
      <Pill color="blue" className="foo" invertTextColor>
        Blue
      </Pill>
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
