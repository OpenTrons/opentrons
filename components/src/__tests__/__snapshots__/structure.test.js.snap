// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders Card correctly 1`] = `
.c0 {
  font-size: 0.875rem;
  position: relative;
  overflow: visible;
  box-shadow: 0 1px 2px 0 rgba(0,0,0,0.33);
}

.c1 {
  font-size: 1.125rem;
  font-weight: 600;
  color: #4a4a4a;
  font-weight: 400;
  margin: 0;
  padding: 1rem 1rem 0;
  text-transform: capitalize;
}

<section
  className="c0"
>
  <h3
    className="c1"
  >
    title
  </h3>
  children children children
</section>
`;

exports[`LabeledValue renders LabeledValue correctly 1`] = `
<div
  className="labeled_value"
>
  <p
    className="labeled_value_label"
  >
    Label
    :
  </p>
  <p
    className="labeled_value_value"
  >
    Value
  </p>
</div>
`;

exports[`PageTabs renders PageTabs correctly 1`] = `
<nav
  className="page_tabs"
>
  <a
    className="tab_link active_tab_link"
    href="/page1"
    onClick={[Function]}
  >
    <h3
      className="tab_title"
    >
      Page1
    </h3>
  </a>
  <span
    className="tab_link"
    to="/page2"
  >
    <h3
      className="tab_title"
    >
      Page2
    </h3>
  </span>
</nav>
`;

exports[`Pill renders Pill correctly 1`] = `
<span
  className="pill foo"
  style={
    Object {
      "backgroundColor": "blue",
    }
  }
>
  Blue
</span>
`;

exports[`Pill renders Pill correctly with inverted text 1`] = `
<span
  className="pill foo invert_text"
  style={
    Object {
      "backgroundColor": "blue",
    }
  }
>
  Blue
</span>
`;

exports[`Splash renders correctly with custom props 1`] = `
.c0.spin {
  -webkit-animation: GLFYz 0.8s steps(8) infinite;
  animation: GLFYz 0.8s steps(8) infinite;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

<div
  className="splash swag"
>
  <svg
    aria-hidden="true"
    className="sc-AxmLO c0"
    fill="currentColor"
    version="1.1"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2.44444444,20.4 C2.44444444,21.0627417 2.99165197,21.6 3.66666667,21.6 L18.3333333,21.6 C19.008348,21.6 19.5555556,21.0627417 19.5555556,20.4 C19.5555556,20.148 19.47,19.908 19.3355556,19.716 L12.2222222,7.62 L12.2222222,2.4 L9.77777778,2.4 L9.77777778,7.62 L2.66444444,19.716 C2.53,19.908 2.44444444,20.148 2.44444444,20.4 Z M3.66666667,24 C1.64162258,24 0,22.3882251 0,20.4 C0,19.68 0.22,19.008 0.611111111,18.444 L7.33333333,6.972 L7.33333333,4.8 C6.65831864,4.8 6.11111111,4.2627417 6.11111111,3.6 L6.11111111,2.4 C6.11111111,1.0745166 7.20552617,0 8.55555556,0 L13.4444444,0 C14.7944738,0 15.8888889,1.0745166 15.8888889,2.4 L15.8888889,3.6 C15.8888889,4.2627417 15.3416814,4.8 14.6666667,4.8 L14.6666667,6.972 L21.3888889,18.444 C21.78,19.008 22,19.68 22,20.4 C22,22.3882251 20.3583774,24 18.3333333,24 L3.66666667,24 Z M12.2222222,16.8 L13.86,15.192 L16.2188889,19.2 L5.78111111,19.2 L9.03222222,13.668 L12.2222222,16.8 Z M11.6111111,12 C11.9486185,12 12.2222222,12.2686292 12.2222222,12.6 C12.2222222,12.9313708 11.9486185,13.2 11.6111111,13.2 C11.2736038,13.2 11,12.9313708 11,12.6 C11,12.2686292 11.2736038,12 11.6111111,12 Z"
      fillRule="evenodd"
    />
  </svg>
</div>
`;

exports[`Splash renders correctly with no props 1`] = `
.c0.spin {
  -webkit-animation: GLFYz 0.8s steps(8) infinite;
  animation: GLFYz 0.8s steps(8) infinite;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

<div
  className="splash"
>
  <svg
    aria-hidden="true"
    className="sc-AxmLO c0"
    fill="currentColor"
    version="1.1"
    viewBox="0 0 50 50"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M46.79 24.83A21.79 21.79 0 1 1 21.23 3.38l.83 5.31a16.42 16.42 0 1 0 5.87 0l.83-5.31a21.82 21.82 0 0 1 18.03 21.45zm-28.14-3.14c3-4.52 6.34-9.23 6.35-9.23s3.37 4.72 6.35 9.23c2.56 3.9 3.31 8.71.15 12.13a8.76 8.76 0 0 1-6.5 2.83 8.76 8.76 0 0 1-6.46-2.82c-3.19-3.43-2.45-8.24.11-12.14zm0 6.62a5.8 5.8 0 0 0 1.42 3.88 6.29 6.29 0 0 0 4.35 2.16h.45a6.38 6.38 0 0 0 6.59-6.17c.05-1.54-.66-4.47-2.41-5a1.65 1.65 0 0 0-.31-.06 2.93 2.93 0 0 0-2.09.91c-2.17 2.08-4.73 4.46-7.96 4.28z"
      fillRule="evenodd"
    />
  </svg>
</div>
`;

exports[`TitleBar renders TitleBar with back button correctly 1`] = `
.c0.spin {
  -webkit-animation: GLFYz 0.8s steps(8) infinite;
  animation: GLFYz 0.8s steps(8) infinite;
  -webkit-transform-origin: center;
  -ms-transform-origin: center;
  transform-origin: center;
}

<header
  className="title_bar"
>
  <button
    className="button_flat title_button inverted"
    onClick={[Function]}
    title="Back"
    type="button"
  >
    <svg
      aria-hidden="true"
      className="sc-AxmLO c0 icon"
      fill="currentColor"
      version="1.1"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15.429 17.019l-4.242-4.59 4.242-4.59-1.312-1.41-5.545 6 5.545 6z"
        fillRule="evenodd"
      />
    </svg>
    back
  </button>
  <h1
    className="title right"
  >
    foo
  </h1>
  <span
    className="separator"
  >
    |
  </span>
  <h2
    className="subtitle"
  >
    bar
  </h2>
</header>
`;

exports[`TitleBar renders TitleBar with subtitle correctly 1`] = `
<header
  className="title_bar"
>
  <h1
    className="title"
  >
    foo
  </h1>
  <span
    className="separator"
  >
    |
  </span>
  <h2
    className="subtitle"
  >
    bar
  </h2>
</header>
`;

exports[`TitleBar renders TitleBar without subtitle correctly 1`] = `
<header
  className="title_bar"
>
  <h1
    className="title"
  >
    foo
  </h1>
</header>
`;
